src/Pos/Explorer/Web/ClientTypes.hs

```diff
@@ -253,10 +253,10 @@ instance FromHttpApiData CHash where
     parseUrlPiece = fmap toCHash . decodeHashHex
 
 instance FromHttpApiData CAddress where
-    parseUrlPiece = fmap toCAddress . decodeTextAddress
+    parseUrlPiece = pure . CAddress
```

Comment:
The (curl) test is in the comments, it works. But I went to double-check:
```
newtype CAddress = CAddress Text
    deriving (Show, Eq, Generic, Buildable, Hashable)

getAddressSummary :: ExplorerMode m => CAddress -> m CAddressSummary
getAddressSummary cAddr = cAddrToAddr cAddr ...

cAddrToAddr :: MonadThrow m => CAddress -> m Address
cAddrToAddr cAddr =
    fromCAddress cAddr &
    either (const $ throwM $ Internal "Invalid address!") pure
```

So there is a validation that checks if the sent `Text` is valid and you can see that the test works in the curl requests.